src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -96,26 +96,20 @@ private static bool FirstCharInUInt32IsAscii(uint value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static unsafe nuint GetIndexOfFirstNonAsciiByte(byte* pBuffer, nuint bufferLength)
         {
-            // If SSE2 is supported, use those specific intrinsics instead of the generic vectorized
-            // code below. This has two benefits: (a) we can take advantage of specific instructions like
-            // pmovmskb which we know are optimized, and (b) we can avoid downclocking the processor while
+            // If 256/512-bit aren't supported but SSE2 is supported, use those specific intrinsics instead of
+            // the generic vectorized code. This has two benefits: (a) we can take advantage of specific instructions
+            // like pmovmskb which we know are optimized, and (b) we can avoid downclocking the processor while
             // this method is running.
 
-            if (Vector512.IsHardwareAccelerated || Vector256.IsHardwareAccelerated)
-            {
-                return GetIndexOfFirstNonAsciiByte_Vector(pBuffer, bufferLength);
-            }
-            else if (Sse2.IsSupported || (AdvSimd.IsSupported && BitConverter.IsLittleEndian))
+            if (!Vector512.IsHardwareAccelerated &&
+                !Vector256.IsHardwareAccelerated &&
+                (Sse2.IsSupported || (AdvSimd.IsSupported && BitConverter.IsLittleEndian)))
```

Comment:
the last one can't be just `Vector128.IsHardwareAccelerated` because of big endian on arm? I'd guess that we already have a few (or more) paths where we don't check endianess because there are no LE targets with SIMD support in .net?

Reply:
This is just the existing condition that was there. But from what Jan said last night, it's useless, so yeah, we can delete the condition.

Reply:
Why?

Reply:
The conditions [before was](https://github.com/dotnet/runtime/pull/90390/files#diff-6b4906abc01dc4699f348f7c1df72e2f640f240aa31ea67cd47642221b2021f5L108)  `else if (Sse2.IsSupported || (AdvSimd.IsSupported && BitConverter.IsLittleEndian))` which does indeed verify `IsLittleEndian`, but the code [now](https://github.com/dotnet/runtime/pull/90390/files#diff-6b4906abc01dc4699f348f7c1df72e2f640f240aa31ea67cd47642221b2021f5R106) is ` (Sse2.IsSupported || AdvSimd.IsSupported)` which doesn't examine the `IsLittleEndian` as the `Debug.Assert`'s comments assume.

Note, the code at the top of this comment thread is what you originally posted, not the current PR state
<img width="447" alt="image" src="https://github.com/dotnet/runtime/assets/874059/1934fd0d-692a-4f28-91c7-9271252ceabe">

Reply:
We removed the IsLittleEndian condition not because it wasn't relevant but because there's no supported platform where AdvSimd.IsSupported would be true but IsLittleEndian would be false. A Debug.Assert helps validate then the assumption being made.